Composer/packages/adaptive-form/src/components/fields/IntellisenseFields.tsx

```diff
@@ -0,0 +1,173 @@
+        <StringField
+          {...props}
+          focused={focused}
+          id={id}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
+          onChange={(newValue) => onValueChanged(newValue || '')}
+          onClick={onClickTextField}
+          onKeyDown={onKeyDownTextField}
+          onKeyUp={onKeyUpTextField}
+        />
+      )}
+    </Intellisense>
+  );
+};
+
+export const IntellisenseExpressionField: React.FC<FieldProps<string>> = function IntellisenseExpressionField(props) {
+  const { id, value = '', onChange, focused: defaultFocused } = props;
+
+  return (
+    <Intellisense
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={['expressions', 'user-variables']}
+      url={getIntellisenseUrl()}
+      value={value}
+      onBlur={props.onBlur}
+      onChange={onChange}
+    >
+      {({ textFieldValue, focused, onValueChanged, onKeyDownTextField, onKeyUpTextField, onClickTextField }) => (
+        <StringField
+          {...props}
+          focused={focused}
+          id={id}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
+          onChange={(newValue) => onValueChanged(newValue || '')}
+          onClick={onClickTextField}
+          onKeyDown={onKeyDownTextField}
+          onKeyUp={onKeyUpTextField}
+        />
+      )}
+    </Intellisense>
+  );
+};
+
+export const IntellisenseNumberField: React.FC<FieldProps<string>> = function IntellisenseNumberField(props) {
+  const { id, value = '', onChange, uiOptions, focused: defaultFocused } = props;
+
+  const completionListOverrideResolver = (value: string) => {
+    if (value === '') {
+      return <ExpressionSwitchWindow type={'Number'} onSwitchToExpression={() => onChange('=')} />;
+    } else {
+      return null;
+    }
+  };
+
+  return (
+    <Intellisense
+      completionListOverrideResolver={completionListOverrideResolver}
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={uiOptions.intellisenseScopes || []}
+      url={getIntellisenseUrl()}
+      value={value}
+      onBlur={props.onBlur}
+      onChange={onChange}
+    >
+      {({ textFieldValue, focused, onValueChanged, onKeyDownTextField, onKeyUpTextField, onClickTextField }) => (
+        <NumberField
+          {...props}
+          focused={focused}
+          id={id}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
+          onChange={(newValue) => onValueChanged(newValue || 0)}
+          onClick={onClickTextField}
+          onKeyDown={onKeyDownTextField}
+          onKeyUp={onKeyUpTextField}
+        />
+      )}
+    </Intellisense>
+  );
+};
+
+export const IntellisenseJSONField: React.FC<FieldProps<string>> = function IntellisenseJSONField(props) {
+  const { id, value = '', onChange, focused: defaultFocused, schema } = props;
+
+  const completionListOverrideResolver = (value: any) => {
```

Comment:
Nit:  You are overriding the completionListResolver so it would probably be better named completionListResolverOverride or overrideCompletionListResolver.  This makes it sound like it is resolving completionListOverrides rather than overriding completionListResolvers :)

Reply:
It is actually intended. Based on the current value the _**completionListOverride**_ is constructed as such: 
![image](https://user-images.githubusercontent.com/66701106/96322155-d2d74380-0fcc-11eb-9658-b6f4c1c5fe97.png)